src/entities/trades/swapr-v3/SwaprV3.ts

```diff
@@ -194,21 +194,28 @@ export class SwaprV3Trade extends TradeWithSwapTransaction {
   }
 
   public async swapTransaction(options: TradeOptions): Promise<UnsignedTransaction> {
-    const to: string = validateAndParseAddress(options.recipient)
-    const amountIn: string = `0x${this.maximumAmountIn().raw.toString(16)}`
-    const amountOut: string = `0x${this.minimumAmountOut().raw.toString(16)}`
+    const isNativeIn = Currency.isNative(this.inputAmount.currency)
+    const isNativeOut = Currency.isNative(this.outputAmount.currency)
+    invariant(!(isNativeIn && isNativeOut), 'the router does not support both native in and out')
```

Comment:
Let's improve this message

```suggestion
    invariant(!(isNativeIn && isNativeOut), 'SwaprV3Trade.swapTransaction: the router does not support both native in and out')
```

Reply:
done